components/base/wheeled/wheeled_base.go

```diff
@@ -115,6 +133,96 @@ func (base *wheeledBase) Spin(ctx context.Context, angleDeg, degsPerSec float64,
 	rpm, revolutions := base.spinMath(angleDeg, degsPerSec)
 
 	return base.runAll(ctx, -rpm, revolutions, rpm, revolutions)
+
+}
+
+func (base *wheeledBase) spinWithMovementSensor(ctx context.Context, angleDeg, degsPerSec float64, extra map[string]interface{}) error {
+	wheelrpm, revs := base.spinMath(angleDeg, degsPerSec)
+
+	currYaw, err := getCurrentYaw(ctx, base.orientationSensor, extra) // from -179 to 179
+	if err != nil {
+		return err
+	}
+
+	errCounter := 0
+	targetYaw := calculatedDomainLimitedAngleError(angleDeg, currYaw)
+	timer := time.NewTicker(time.Duration(pollTimeMs * float64(time.Millisecond))) //~rename
+	for {
+
+		select {
```

Comment:
[q] why do we need this original select statement, if we're checking the same condition immediately below?

Reply:
I'm still tooling with the physical performance for the PR and seeing if we can get within 15 deg error. Which is why these both exist. Either the default select case will remain without a timer or the timer logic will remain.

Reply:
keeping this open till I test again.